File: Python/Problem 3.py
class Solution(object):
    def lengthOfLongestSubstring(s) -> int:
        charMap = {}
        res = 0
        slide = 0

        for end, c in enumerate(s):
            if c in charMap and charMap[c] >= slide:
                slide = charMap[c] + 1
            charMap[char] = end
            res = max(res, end-slide+1)

File: TypeScript/Problem 3.ts
function lengthOfLongestSubstring(s: string): number {
    const charMap: {[key: char]: number} = {};
    let res = 0;
    let slide = 0;

    s = s.toLowercase()

    for (const c of s){
        if (c in charMap && charMap[c] >= slide){
            slide = charMap[c]+1;
        }
        let endSlide = s.indexOf(c, start);
        charMap[c] = s.indexOf(c, slide);
        let currentWindow = endSlide-Slide+1;
        res = Math.max(res, currentWindow);
    }

    return res;
};
